### PR TITLE
feature/cplp- 829 get all language

### PR DESCRIPTION
**Pull request by qxz2mqe**
https://jira.catena-x.net/browse/CPLP-829

### CODE REVIEW

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Models/LanguageData.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+/// </summary>
+public class LanguageData
+{
+    /// <summary>
+    /// Constructor.
+    /// </summary>
+    /// <param name="languageShortName">Language Short Name</param>
+    /// <param name="longNameDe">Language Long Name in German</param>
+    /// <param name="longNameEn">Language Long Name in English</param>
+    public LanguageData(string languageShortName, string longNameDe, string longNameEn)
```
Comment (Phil91) [inline src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Models/LanguageData.cs:36]: same idea as in #132 to change the class to an record

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/StaticDataRepository.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    /// <inheritdoc />
+    public IAsyncEnumerable<UseCaseData?> GetAllUseCase()
+    =>
+        _dbContext.UseCases
+            .AsNoTracking()
+            .Select(s => new UseCaseData(s.Id, s.Name, s.Shortname))
+            .ToAsyncEnumerable();
+    
+    /// <inheritdoc />
+    public IAsyncEnumerable<LanguageData?> GetAllLanguage()
```
Comment (Phil91) [inline src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/StaticDataRepository.cs:51]: Same as in #132 i don't think the LanguageData needs to be nullable

File: src/administration/CatenaX.NetworkServices.Administration.Service/Controllers/StaticDataController.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        _logic.GetAllUseCase();
+    
+    /// <summary>
+    /// Retrieves all Language
+    /// </summary>
+    /// <returns>AsyncEnumerable of Language Data</returns>
+    /// <remarks>
+    /// Example: GET: /api/administration/staticdata/languagetags
+    /// </remarks>
+    /// <response code="200">Returns a list of all of the Language data.</response>
```
Comment (jjeroch) [inline src/administration/CatenaX.NetworkServices.Administration.Service/Controllers/StaticDataController.cs:71]: @qxz2mqe small remark. I suggest we add here or inside the summary the information, that the "lang" parameter is an optional parameter and if not set "en" will be used

Comment (jjeroch) [inline src/administration/CatenaX.NetworkServices.Administration.Service/Controllers/StaticDataController.cs:71]: @qxz2mqe  we should include the information that we return german and english.....either here or in the API summary.

File: src/administration/CatenaX.NetworkServices.Administration.Service/Controllers/StaticDataController.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    /// <response code="200">Returns a list of all of the use case data.</response>
+    [HttpGet]
+    [Authorize(Roles = "view_use_cases")]
+    [Route("usecases")]
+    [ProducesResponseType(typeof(IAsyncEnumerable<UseCaseData>), StatusCodes.Status200OK)]
+    public IAsyncEnumerable<UseCaseData> GetUseCases() =>
+        _logic.GetAllUseCase();
+    
+    /// <summary>
+    /// Retrieves all Language
```
Comment (jjeroch) [inline src/administration/CatenaX.NetworkServices.Administration.Service/Controllers/StaticDataController.cs:65]: Summary to be changed to "Retrieve all app language tags - short name (2digit) and long name"

Comment (qxz2mqe) [reply src/administration/CatenaX.NetworkServices.Administration.Service/Controllers/StaticDataController.cs]: fixed

File: src/administration/CatenaX.NetworkServices.Administration.Service/Controllers/StaticDataController.cs (hunk 3 above)
Comment (qxz2mqe) [reply src/administration/CatenaX.NetworkServices.Administration.Service/Controllers/StaticDataController.cs]: fixed

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/StaticDataRepository.cs (hunk 2 above)
Comment (qxz2mqe) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/StaticDataRepository.cs]: fixed

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Models/LanguageData.cs (hunk 1 above)
Comment (qxz2mqe) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Models/LanguageData.cs]: fixed

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**